src/linear_samplers/block_col_count_sketch.jl

```diff
@@ -0,0 +1,86 @@
+# This file is part of RLinearAlgebra.jl
+# 1. Specifies type
+# 2. Implements sample function
+# 3. Exports type
+#
+# Date: 07/19/2024
+# Author: Christian Varner
+# Purpose: Implement the column sketching algorithm called CountSketch.
+
+"""
+    LinSysBlkColCountSketch <: LinSysVecColSelect
+
+A mutable structure that represents the CountSketch algorithm for columns.
+The assumption is that `A` is fully known (that is, the sampling procedure
+is not used in a streaming context).
+
+See Kenneth L. Clarkson and David P. Woodruff. 2017. 
+    "Low-Rank Approximation and Regression in Input Sparsity Time."
+    J. ACM 63, 6, Article 54 (February 2017), 45 pages. 
+    https://doi.org/10.1145/3019134
+
+# Fields
+
+- `block_size::Int64`, is the number of columns in the sketched matrix `A * S`
+- `S::Union{Matrix{Int64}, Nothing}`, buffer matrix for storing the sampling matrix `S`.
+- `signs::Union{Vector{Int64}, Nothing}`, buffer vector for storing data used in `sample`.
+
+Additional Constructors:
+
+Calling `LinSysBlkColCountSketch(block_size)` defaults to `LinSysBlkColCountSketch(block_size, nothing, nothing)`.
+Calling `LinSysBlkColCountSketch()` defaults to `LinSysBlkColCountSketch(2, nothing, nothing)`. 
+
+Remark: Current implementation does not take advantage of sparse matrix data structures or operations.
```

Comment:
@nathanielpritchard are we using `!!!` for these types of situations? https://documenter.juliadocs.org/stable/showcase/#Admonitions ?